### PR TITLE
Upgrade nvidia-container-toolkit to fix CVE-2024-0132 CVE-2024-0133

### PR DESCRIPTION
**Pull request by mandeepsplaha**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
This PR fixes  to fix critical CVE-2024-0132 and medium CVE-2024-0133 in `nvidia-container-toolkit` by upgrading the version to `1.16.2`.

###### Change Log  

- CVE-2024-0132
- CVE-2024-0133

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2024-0132
- https://nvd.nist.gov/vuln/detail/CVE-2024-0133

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=653725&view=results

### CODE REVIEW

**Review by jslobodzian — APPROVED** [review]

**Comment by henryli001** [comment]
Since you have upgraded nvidia-container-toolkit, you also need to upgrade the libnvidia-container package. The version of these two packages have to match or else the containerized GPU scenario will break.

**Review by henryli001 — CHANGES_REQUESTED** [review]
Since you have upgraded nvidia-container-toolkit, you also need to upgrade the libnvidia-container package. The version of these two packages have to match or else the containerized GPU scenario will break.

**Comment by PawelWMS** [comment]
> Since you have upgraded nvidia-container-toolkit, you also need to upgrade the libnvidia-container package. The version of these two packages have to match or else the containerized GPU scenario will break.

@henryli001, since we're on the topic, do you know if there's are reason these two packages weren't added to the "Spec Entanglement Mismatch Check"? It sounds like it would help keeping these package in-sync.

**Comment by henryli001** [comment]
> > Since you have upgraded nvidia-container-toolkit, you also need to upgrade the libnvidia-container package. The version of these two packages have to match or else the containerized GPU scenario will break.
> 
> @henryli001, since we're on the topic, do you know if there's are reason these two packages weren't added to the "Spec Entanglement Mismatch Check"? It sounds like it would help keeping these package in-sync.

I'm not sure about the actual details of why these packages were not included but I think they should.

File: SPECS/libnvidia-container/libnvidia-container.spec [hunk 1]
```diff
@@ -3,7 +3,7 @@
 %define mod_probe_dir deps/src/nvidia-modprobe-%{modprobe_version}
 Summary:        NVIDIA container runtime library
 Name:           libnvidia-container
-Version:        1.15.0
+Version:        1.16.2
```
Comment (mfrw) [inline SPECS/libnvidia-container/libnvidia-container.spec:6]: Fixes: https://github.com/microsoft/azurelinux/pull/10636

**Review by henryli001 — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Pull request merged**